R/fotmob_leagues.R

```diff
@@ -228,28 +248,65 @@ fotmob_get_league_matches <- function(country, league_name, league_id, cached =
     league_id = rlang::maybe_missing(league_id, NULL)
   )
 
-  fp <- purrr::possibly(
-    .fotmob_get_league_matches,
-    quiet = FALSE,
-    otherwise = tibble::tibble()
+  # Need to coerce to `NA_character` since crossing doesn't like `NULL`
+  season <- ifelse(is.null(season), NA_character_, season)
+  urls <- tidyr::crossing(
+    urls,
+    "season" = season
   )
-  purrr::map2_dfr(
-    urls$id, urls$page_url,
+
+  purrr::pmap_dfr(
+    list(
+      urls$id,
+      urls$page_url,
+      urls$season
+    ),
     .fotmob_get_league_matches
   )
 }
 
+#' @importFrom glue glue glue_collapse
+#' @importFrom rlang inform
+.fotmob_message_for_season <- function(resp, season = NULL) {
+
+  if (is.null(season)) {
+    season <- resp$allAvailableSeasons[1]
+    rlang::inform(
+      glue::glue('Defaulting `season` to latest ("{season}").'),
+      .frequency = "once",
+      .frequency_id = ".fotmob_get_league_(matches|tables)"
+    )
+  } else {
+    if (isFALSE(season %in% resp$allAvailableSeasons)) {
+      stop(
+        glue::glue(
+          "`season` should be one of the following:\n{glue::glue_collapse(resp$allAvailableSeasons, '\n')}"
+        )
+      )
+    }
+  }
+
+}
+
+
 #' @importFrom janitor clean_names
 #' @importFrom tibble as_tibble
 #' @importFrom purrr map_dfr
 #' @importFrom dplyr bind_rows
-.fotmob_get_league_matches <- function(league_id, page_url) {
-  resp <- .fotmob_get_league_resp(league_id, page_url)
+.fotmob_get_league_matches <- function(league_id, page_url, season = NULL) {
+  # And now coerce NA back to NULL
+  season <- switch(!is.na(season), season, NULL)
```

Comment:
have to use `switch()` instead of `if` or `ifelse()` since those suck with handling `NULL`s